### ISSUE TITLE
Config Doesn't Work: Expects Project Name and Namespace

### ISSUE THREAD

**Issue by rawkode** [pulumi/pulumi#3223]
Using Pulumi `1.1.0`

Created a stack with `pulumi stack init production --secrets-provider="gcpkms: ...`
Selected stack with `pulumi stack select production`
Set some config with `pulumi config set telegraf:version "1.12"`

`Pulumi.production.yaml` looks like:

```yaml
secretsprovider: gcpkms:...
encryptedkey: secret-key
config:
  telegraf:version: "1.12"
```

Inside my code, I have:

```typescript
const data = {
  telegrafVersion: config.require("telegraf:version"),
}
```

Running a `pulumi up` responds with:

```
 pulumi up
Previewing update (production):

     Type                 Name                       Plan       Info
 +   pulumi:pulumi:Stack  devrel-metrics-production  create     1 error
 
Diagnostics:
  pulumi:pulumi:Stack (devrel-metrics-production):
    error: Missing required configuration variable 'devrel-metrics:telegraf:version'
        please set a value using the command `pulumi config set devrel-metrics:telegraf:version <value>`
```

**Comment by rawkode** [comment]
NB: I did try the config set command with the project name too:

```
╰ pulumi config set devrel-metrics:telegraf:version "1.12"
error: invalid configuration key: could not parse devrel-metrics:telegraf:version as a configuration key (configuration keys should be of the form `<namespace>:<name>`)
```

**Comment by ellismg** [comment]
I'm guessing that `config` here is a local which you have assigned to the result of `new pulumi.Config()`.  The way the config system works is that values set as `<namespace>:<name>`.  If you set a value on the command line that does not have a colon in it, then we assume it is in the namespace for your project (here that's `devrel-metrics`).  Since there is a colon, we assume this value is in the `telegraf` namespace (similar to how if you set `gcp:region` we use `gcp` as the namespace and `region` as the key.

When constructing the `Config` object, if you don't pass a namespace, we use the current project as the namespace.

So in this case, you would want to do something like `(new pulumi.Config("telegraf")).require("version");`

**Comment by nesl247** [comment]
I think he forgot to close this out. It was already resolved in Slack.

**Comment by rawkode** [comment]
Yeah. It wasn't clear to me that the namespace was the project name by default. Closing.

Thanks for your help

**Comment by 1tmb** [comment]
Would be great for https://www.pulumi.com/learn/building-with-pulumi/secrets/ to be expanded to explain this. Found this on a chance google after having the same problem as the OP. Cheers

**Comment by ericrudder** [comment]
@interurban Want to take a look at the doc here?

**Comment by RDavis3000** [comment]
2024 here, spent 2 hours trying to get this to work
